### PR TITLE
Adding subcategories on category page

### LINKED ISSUE #10407: Display subcategories in FO categories pages
After updating from 1.7.2.4 to 1.7.4.2 :   
Subcategories don't display at all (none their thumbnails).

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Before 1.7 there was subcategories displayed on category page
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #10407 
| How to test?  | Go on category page which has subcategories and see if subcategories are right displayed

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16928)

### CODE REVIEW

File: themes/classic/templates/catalog/_partials/subcategories.tpl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
```
Comment (Progi1984) [inline themes/classic/templates/catalog/_partials/subcategories.tpl]:
```suggestion
 * @copyright 2007-2020 PrestaShop SA and Contributors
```

**Review by Progi1984 — COMMENTED** [review]
Small changes

File: themes/classic/templates/catalog/_partials/subcategories.tpl (hunk 1 above)
Comment (PierreRambaud) [reply themes/classic/templates/catalog/_partials/subcategories.tpl]: Not resolved yet

File: themes/classic/templates/catalog/_partials/subcategories.tpl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *}
+{if isset($subcategories) && $subcategories|@count >= 1}
+    {if (isset($display_subcategories) && $display_subcategories eq 1) || !isset($display_subcategories) }
```
Comment (PierreRambaud) [inline themes/classic/templates/catalog/_partials/subcategories.tpl]: should be 2 spaces for indentation, no?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
I rebased on develop and removed the asset merged some days ago!

Need review again before QA

File: themes/classic/templates/catalog/_partials/subcategories.tpl [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{**
+ * 2007-2019 PrestaShop and Contributors
```
Comment (Progi1984) [inline themes/classic/templates/catalog/_partials/subcategories.tpl]: Please update the license header

File: themes/classic/templates/catalog/_partials/subcategories.tpl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *}
+{if isset($subcategories) && $subcategories|@count >= 1}
```
Comment (Progi1984) [inline themes/classic/templates/catalog/_partials/subcategories.tpl]:
```suggestion
{if isset($subcategories) && $subcategories is not empty}
```

**Review by Progi1984 — COMMENTED** [review]
Two feedbacks

File: themes/classic/templates/catalog/_partials/subcategories.tpl (hunk 4 above)
Comment (PierreRambaud) [inline themes/classic/templates/catalog/_partials/subcategories.tpl]: It's not a twig line
```suggestion
{if !empty($subcategories)}
```

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by SD1982** [comment]
This pull request correctly display sub-category on main category page . Thanks @NeOMakinG

**Comment by Progi1984** [comment]
Thanks @NeOMakinG

**Pull request merged**